extensions/amp-list/0.1/amp-list.js

```diff
@@ -340,6 +349,56 @@ export class AmpList extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Returns true if element's src points to amp-state.
+   *
+   * @param {string} src
+   * @return {boolean}
+   * @private
+   */
+  isAmpStateSrc_(src) {
+    return (
+      isExperimentOn(this.win, 'amp-list-init-from-state') &&
+      startsWith(src, AMP_STATE_URI_SCHEME)
+    );
+  }
+
+  /**
+   * Render an amp-list that has an "amp-state:" uri. For example,
+   * src="amp-state:json.path".
+   *
+   * @param {string} src
+   * @return {Promise<void>}
+   * @private
+   */
+  renderAmpStateJson_(src) {
```

Comment:
Can't this be moved into `fetchList_`? Refactoring it so that it only does the fecthing (either from network, SSR, or amp-state). That'll reduce the duplication in the handling logic after the fetch.

Reply:
I had it that way initially, but after realizing that I'd prefer `amp-state: ` URIs to only be valid on initial load, I thought this was cleaner. Heres the commit where I moved it from `fetchList_` to its own func -- I'm happy to revert if you prefer it!

commit: https://github.com/ampproject/amphtml/pull/26284/commits/abcc558695178a39271cac0237958ed356ed99aa

Reply:
I've updated this based on our offline conversation. I've modified the expectation of `amp-state:` to be pointed at equivalent JSON to the return of what the xhr would have been.

Also, I noticed an unrelated failing test, which I've spun out into its own PR: https://github.com/ampproject/amphtml/pull/26509